Enrich_methods/material_db_tools.py

```diff
@@ -0,0 +1,444 @@
+from typing import Dict, Union, Optional
+from pyne import material
+from pyne.material import Material, MultiMaterial
+from pyne.material_library import MaterialLibrary
+from pyne import nucname
+from decimal import Decimal, getcontext
+
+
+def make_mat(
+    nucvec: Dict[int, float],
+    density: float,
+    citation: str,
+    molecular_mass: Optional[float] = None,
+    mass_enrichment: Optional[Dict[int, Dict[int, float]]] = None,
+) -> Material:
+    """
+    Create a Material object from nuclear vector data.
+
+    Args:
+        nucvec (Dict[int, float]): The nuclear mass composition vector, with keys as nuclide IDs (zas)
+            or the pyne specified element name and values as their fractions (float).
+        density (float): The density of the material.
+        citation (str): A citation or reference for the material data.
+        molecular_mass (Optional[float]): The molecular mass of the material, if applicable.
+        mass_enrichment (Optional[Dict[int, Dict[int, float]]]): A dictionary containing the
+            enrichment information for elements, where keys are element IDs (int) and values are
+            dictionaries with nuclide IDs (int) as keys and enrichment fractions (float) as values.
+
+    Returns:
+        Material: The created Material object with the specified properties, based on mass fractions.
+    """
+    if mass_enrichment:
+        for element, enrichment_vector in mass_enrichment.items():
+            enriched_mat = Material(enrichment_vector)
+            nucvec = enrich(nucvec, element, enriched_mat.comp)
+    mat = Material(nucvec, density=density, metadata={"citation": citation})
+    if molecular_mass:
+        mat.molecular_mass = molecular_mass
+    return mat.expand_elements()
+
+
+def make_mat_from_atom(
+    atom_frac: Dict[Union[int, str], float],
+    density: float,
+    citation: str,
+    mass_enrichment: Optional[Dict[int, Dict[int, float]]] = None,
```

Comment:
Should there also be a version that takes an atom_enrichment? or is that too rare and we just make users work around it?